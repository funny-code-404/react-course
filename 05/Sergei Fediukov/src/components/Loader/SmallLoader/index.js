import loading from './Loading.gif'
const SmallLoader = () => {
    return (
        <div className='small_loader'>
            <img src={loading} alt="loading..." />
        </div>
    )
}
export default SmallLoader
