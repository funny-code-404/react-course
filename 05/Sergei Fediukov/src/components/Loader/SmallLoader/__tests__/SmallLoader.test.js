import { configure, shallow } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'

import SmallLoader from '..'

configure({ adapter: new Adapter() });

describe('Should render correctly "Civilizations"', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<SmallLoader />)
    })
    it('to Match Snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })
    it('component is defined', () => {
        expect(wrapper).toHaveLength(1)
    })
    it('component have children div with "small_loader" class', () => {
        expect(wrapper.find('.small_loader')).toHaveLength(1)
    })
    it('component have children img', () => {
        expect(wrapper.find('img')).toHaveLength(1)
    })
})