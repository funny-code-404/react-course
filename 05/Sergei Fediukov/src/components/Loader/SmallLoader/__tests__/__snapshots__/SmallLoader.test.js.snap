// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly "Civilizations" to Match Snapshot 1`] = `
<div
  className="small_loader"
>
  <img
    alt="loading..."
    src="Loading.gif"
  />
</div>
`;
